server/athenian/api/controllers/reposet.py

```diff
@@ -50,7 +55,16 @@
     rs = await db.fetch_one(select(columns).where(RepositorySet.id == id))
     if rs is None or len(rs) == 0:
         raise ResponseError(NotFoundError(detail="Repository set %d does not exist" % id))
-    if rs[RepositorySet.owner.key] != user.id:
+    if account is None:
+        # if there is no account, suppose the reposet owner
+        account = rs[RepositorySet.owner.key]
```

Comment:
Shouldn't it be the `account` the `user` is associated with? So something like:
```
if account is None:
    account = <get account from user>

if rs[RepositorySet.owner.key] != account:
   ...
```

Reply:
This is equivalent, I think. The code below is

```python
status = await db.fetch_one(select([UserAccount.is_admin]).where(
        and_(UserAccount.user_id == user.id, UserAccount.account_id == account)))
```

So we check the user's access to the reposet in SQL. Taking an external `account` is a convenience for the metrics endpoint which ensures that the user does not mix reposets from different accounts together.

I should remove the `account` arg completely and check the reposet mixture on the calling side.